### PR TITLE
Feature/extra attributes

### PR DESCRIPTION
**Pull request by rfocosi**
Added extra functionalities to TriggeredSend. It helps sending attributes with spaces and sending the API calls asynchronously.

As wrote on README:

- If you have DataExtension's keys with spaces or any unusual pattern

```ruby
external_key = 'My TriggeredSend External Key'
ts = ExactTargetRest::TriggeredSend.new(auth, external_key)
ts.with_options(email_address: 'uga@kabuga.com', subscriber_attributes: { 'An Attribute' => 'XXX', 'Another_Attribute' => 'YYY' }).deliver
```

- if you have to call the api asynchronously

```ruby
def deliver
  external_key = 'My TriggeredSend External Key'
  triggered_send = ExactTargetRest::TriggeredSend.new(auth, external_key)
  triggered_send.with_options(email_address: 'uga@kabuga.com')

  Worker.perform_async(triggered_send.to_yaml)
end

class Worker
  include Sidekiq::Worker

  sidekiq_options queue: :exact_target_mailer

  def perform(triggered_send)
    YAML::load(triggered_send).deliver
  end
end

```

### CODE REVIEW

File: lib/exact_target_rest/authorization.rb [hunk 1]
```diff
@@ -21,31 +21,30 @@ def initialize(client_id, client_secret)
...
       end
       if resp.success?
         @access_token = resp.body['accessToken']
         @expires_in = Time.now + resp.body['expiresIn']
+        { access_token: @access_token, expires_in: @expires_in }
       else
-        fail resp.body.inspect
+        fail NotAuthorizedError
       end
+    rescue NotAuthorizedError
```
Comment (ruliana) [inline lib/exact_target_rest/authorization.rb]: Maybe a "raise" to rethrows exception after "retry"? Otherwise I guess the Exception is eaten =/

Comment (rfocosi) [inline lib/exact_target_rest/authorization.rb]: Maybe this retry should not be here. It should be implemented by the caller. What you think about removing this retry?

Comment (rfocosi) [inline lib/exact_target_rest/authorization.rb]: And the caller is not the `with_authorization`, the real caller :)

Comment (ruliana) [inline lib/exact_target_rest/authorization.rb]: Hmmmm...

Let's leave that as is for now. We can change it later :)

**Pull request merged**